app/models/katello/repository.rb

```diff
@@ -29,6 +29,7 @@ class Repository < Katello::Model
     DOCKER_TYPE = 'docker'.freeze
     OSTREE_TYPE = 'ostree'.freeze
     ANSIBLE_COLLECTION_TYPE = 'ansible_collection'.freeze
+    GENERIC_CONTENT_TYPE = 'generic_content_unit'.freeze
```

Comment:
I'm thinking you might want to name the repository type `generic` instead of `generic_content_unit`.   `generic_content_unit` sounds too content-unit-type-focused to me.  If you're following the file example, file's repo type is `file` and its content type is `file_unit`.